AutomatedTesting/Editor/Scripts/scene_mesh_to_prefab.py

```diff
@@ -0,0 +1,147 @@
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+import os, traceback, binascii, sys, json, pathlib
+import azlmbr.math
+import azlmbr.bus
+
+#
+# SceneAPI Processor
+#
+
+
+def log_exception_traceback():
+    exc_type, exc_value, exc_tb = sys.exc_info()
+    data = traceback.format_exception(exc_type, exc_value, exc_tb)
+    print(str(data))
+
+def raise_error(message):
+    print (f'ERROR - {message}');
+    raise RuntimeError(f'[ERROR]: {message}');
+
+def get_mesh_node_names(sceneGraph):
+    import azlmbr.scene as sceneApi
+    import azlmbr.scene.graph
+    from scene_api import scene_data as sceneData
+
+    meshDataList = []
+    node = sceneGraph.get_root()
+    children = []
+    paths = []
+
+    while node.IsValid():
+        # store children to process after siblings
+        if sceneGraph.has_node_child(node):
+            children.append(sceneGraph.get_node_child(node))
+
+        nodeName = sceneData.SceneGraphName(sceneGraph.get_node_name(node))
+        paths.append(nodeName.get_path())
+
+        # store any node that has mesh data content
+        nodeContent = sceneGraph.get_node_content(node)
+        if nodeContent.CastWithTypeName('MeshData'):
+            if sceneGraph.is_node_end_point(node) is False:
+                if (len(nodeName.get_path())):
+                    meshDataList.append(sceneData.SceneGraphName(sceneGraph.get_node_name(node)))
+
+        # advance to next node
+        if sceneGraph.has_node_sibling(node):
+            node = sceneGraph.get_node_sibling(node)
+        elif children:
+            node = children.pop()
+        else:
+            node = azlmbr.scene.graph.NodeIndex()
+
+    return meshDataList, paths
+
+def update_manifest(scene):
+    import json
+    import uuid, os
+    import azlmbr.scene as sceneApi
+    import azlmbr.scene.graph
+    from scene_api import scene_data as sceneData
+
+    graph = sceneData.SceneGraph(scene.graph)
+    mesh_name_list, all_node_paths = get_mesh_node_names(graph)
+    scene_manifest = sceneData.SceneManifest()
+    source_basepath = scene.watchFolder
+
+    clean_filename = scene.sourceFilename.replace('.', '_')
```

Comment:
Is this enough to sanitize a filename? What about other illegal filename characters like `[`, '\', '|', etc ?

Reply:
the names coming into this script are from Maya or Blender where those characters would not exist.